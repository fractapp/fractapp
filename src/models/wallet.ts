import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export function getLogo(currency: Currency) {
    let logo: ImageSourcePropType
    switch (currency) {
        case Currency.Polkadot:
            logo = require("assets/img/dot.png")
            break
        case Currency.Kusama:
            logo = require("assets/img/kusama.png")
            break
        default:
            throw ("invalid currency")
    }
    return logo
}
export function getSymbol(currency: Currency) {
    let symbol = ""
    switch (currency) {
        case Currency.Polkadot:
            symbol = "DOT"
            break
        case Currency.Kusama:
            symbol = "KSM"
            break
        default:
            throw ("invalid currency")
    }
    return symbol
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;
    usdValue: number;

    constructor(name: string, address: string, currency: Currency, balance: number, price: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;

        this.usdValue = +(balance * price).toFixed(2)
    }
}
