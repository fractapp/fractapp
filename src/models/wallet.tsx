import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;

    symbol: string;
    usdValue: number;
    logo: ImageSourcePropType;

    constructor(name: string, address: string, currency: Currency, balance: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;
        
        //TODO get state
        let rate = 10
        this.usdValue = balance * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
