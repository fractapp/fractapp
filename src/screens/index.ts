export {ImportWallet} from './ImportWallet';
export {SaveSeed} from './SaveSeed';
export {SaveWallet} from './SaveWallet';
export {SettingWallet} from './SettingWallet';
export {Start} from './Start';
export {WalletDetails} from './WalletDetails';
export {WalletFileBackup} from './WalletFileBackup';
export {Wallets} from './Wallets';
export {Settings} from './Settings';
export {ConfirmSaveSeed} from './ConfirmSaveSeed';
export {ImportSeed} from './ImportSeed';
export {WalletFileImport} from './WalletFileImport';
export {GoogleDrivePicker} from './GoogleDrivePicker';
export {Receive} from './Receive';
export {Milestone} from './Milestone';
export {TransactionDetails} from './TransactionDetails';
export {SplashScreen} from './SplashScreen';
export {Backup} from './Backup';
export {NewPassCode} from './NewPassCode';
export {VerifyPassCode} from './VerifyPassCode';
export {EditProfile} from './EditProfile';
export {EditPhoneNumber} from './EditPhoneNumber';
export {SelectCountry} from './SelectCountry';
export {ConfirmCode} from './ConfirmCode';
export {EditEmail} from './EditEmail';
export {Navigation} from './Navigation';
export {Chats} from './Chats';
export {Chat} from './Chat';
export {Connecting} from './Connecting';
export {EditUsername} from './EditUsername';
export {EditName} from './EditName';
export {Search} from './Search';
export {SelectWallet} from './SelectWallet';
export {Send} from './Send';
export {EnterAmount} from './EnterAmount';
export {SendWithEnterAddress} from './SendWithEnterAddress';
