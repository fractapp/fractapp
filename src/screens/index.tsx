export { default as ImportWallet } from './ImportWallet';
export { default as SaveSeed } from './SaveSeed';

export { default as SaveWallet } from './SaveWallet';
export { default as SettingWallet } from './SettingWallet';
export { default as Start } from './Start';

export { default as WalletDetails } from './WalletDetails';

export { default as WalletEncryption } from './WalletEncryption';
export { default as Wallets } from './Wallets';
export { ConfirmSaveSeed } from './ConfirmSaveSeed'