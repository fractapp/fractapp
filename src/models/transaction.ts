import {Currency} from './wallet';

/**
 * Transaction type
 * @category Models
 */
export enum TxType {
  None = 0,
  Sent,
  Received,
}

export enum TxStatus {
  Pending = 0,
  Success,
  Fail,
}

/**
 * @category Models
 */
export class Transaction {
  id: string;
  member: string;
  currency: Currency;
  value: number;
  txType: TxType;
  timestamp: number;
  fee: number;
  usdValue: number;
  usdFee: number;
  status: TxStatus;

  constructor(
    id: string,
    member: string,
    currency: Currency,
    txType: TxType,
    timestamp: number,
    value: number,
    usdValue: number,
    fee: number,
    usdFee: number,
    status: TxStatus,
  ) {
    this.id = id;
    this.member = member;
    this.currency = currency;
    this.value = value;
    this.txType = txType;
    this.timestamp = timestamp;
    this.fee = fee;
    this.usdValue = usdValue;
    this.usdFee = usdFee;
    this.status = status;
  }
}
