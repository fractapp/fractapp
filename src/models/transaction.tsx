import { ImageSourcePropType } from "react-native";
import { Currency } from "./wallet";

export enum TxType {
    None = 0,
    Sent,
    Received
}

export class Transaction {
    id: string;
    member: string;
    currency: Currency;
    value: number;
    txType: TxType;
    date: Date;

    usdValue: number;
    symbol: string;
    logo: ImageSourcePropType;

    constructor(id: string, member: string, currency: Currency, txType: TxType, date: Date, value: number) {
        this.id = id;
        this.member = member;
        this.currency = currency;
        this.value = value;
        this.txType = txType;
        this.date = date;
        //TODO get state
        let rate = 10
        this.usdValue = value * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
