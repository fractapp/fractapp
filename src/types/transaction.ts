import {Currency} from './/wallet';

/**
 * Transaction type
 * @category Models
 */
export enum TxType {
  None = 0,
  Sent,
  Received,
}

/**
 * Transaction action type
 * @category Models
 */
export enum TxAction {
  Transfer,
  StakingReward,
  StakingCreateWithdrawalRequest,
  StakingWithdrawn,
  StakingOpenDeposit,
  StakingAddAmount,
  ConfirmWithdrawal,
  UpdateNomination
}

export enum TxStatus {
  Pending = 0,
  Success,
  Fail,
}

/**
 * @category Models
 */
export type Transaction = {
  id: string;
  hash: string;
  userId: string | null;

  address: string;
  currency: Currency;
  action: TxAction,
  txType: TxType;
  timestamp: number;

  value: number;
  planckValue: string;
  usdValue: number;
  fullValue: string;

  fee: number;
  planckFee: string;
  usdFee: number;

  status: TxStatus;
};


/**
 * @category Models
 */
export type ApiTransaction = {
  id: string;
  hash: string;
  currency: Currency;

  from: string;
  userFrom: string;

  action: TxAction,

  to: string;
  userTo: string;

  value: string;
  fee: string;
  price: number

  timestamp: number;
  status: TxStatus;
};

