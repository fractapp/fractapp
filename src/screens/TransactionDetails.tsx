import React, { useState, useContext } from 'react';
import { StyleSheet, View, Image, Text } from 'react-native';
import { Transaction, TxType } from 'models/transaction';
import { getLogo, getSymbol, Wallet } from 'models/wallet';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import { WalletInfo } from 'components/';

export const TransactionDetails = ({ route }: { route: any }) => {
    const tx: Transaction = route.params.transaction
    const wallet: Wallet = route.params.wallet

    return (
        <View style={{ flexDirection: "column", flex: 1, alignItems: "center" }}>
            <View style={styles.info}>
                <Image
                    source={getLogo(tx.currency)}
                    style={styles.logo}
                />
                <Text style={styles.address}>{tx.member}</Text>
                <Text style={styles.value}>{tx.txType == TxType.Sent ? "-" : "+"}{tx.value} {getSymbol(tx.currency)}</Text>
                <View style={styles.status}>
                    <MaterialIcons name="done" size={25} color="#888888" />
                    <Text style={styles.statusText}>Success</Text>
                </View>
            </View>

            <View style={{ width: '100%', marginTop: 30 }}>
                <Text style={[styles.title, { marginBottom: 10 }]}>Write-off account</Text>
                <WalletInfo wallet={wallet} />
            </View>

            <View style={{ width: '100%', flexDirection: 'row', marginTop: 20 }}>
                <View style={{ flex: 1 }}>
                    <Text style={[styles.title, { marginBottom: 5 }]}>Date</Text>
                    <Text style={styles.dateAndFee}>{tx.date.toLocaleDateString()} {tx.date.toLocaleTimeString()}</Text>
                </View>
                <View style={{ flex: 1, alignItems: 'flex-end' }}>
                    <View>
                        <Text style={[styles.title, { marginBottom: 5 }]}>Fee</Text>
                        <Text style={styles.dateAndFee}>{tx.fee} {getSymbol(tx.currency)}</Text>
                    </View>
                </View>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    info: {
        width: '90%',
        marginTop: 20,
        paddingTop: 30,
        paddingBottom: 40,
        borderColor: "#888888",
        borderWidth: 0.5,
        borderRadius: 10,
        alignItems: 'center',
        justifyContent: 'center'
    },
    logo: {
        width: 80,
        height: 80,
        borderRadius: 75,
        overflow: "hidden"
    },
    address: {
        width: '80%',
        marginTop: 20,
        textAlign: 'center',
        fontSize: 16,
        fontFamily: "Roboto-Regular",
        color: "black",
    },
    value: {
        marginTop: 10,
        textAlign: 'center',
        fontSize: 40,
        fontFamily: "Roboto-Regular",
        color: "black",
    },
    status: {
        marginTop: 5,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
    },
    statusText: {
        fontSize: 18,
        fontFamily: "Roboto-Regular",
        color: "#888888",
    },
    title: {
        marginLeft: 20,
        marginRight: 20,
        alignSelf: 'flex-start',
        fontSize: 18,
        fontFamily: "Roboto-Regular",
        color: "#888888",
    },
    dateAndFee: {
        marginLeft: 20,
        marginRight: 20,
        alignSelf: 'flex-start',
        fontSize: 18,
        fontFamily: "Roboto-Regular",
        color: "black",
    },
});