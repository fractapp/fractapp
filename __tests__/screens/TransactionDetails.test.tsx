import React from 'react';
import { TransactionDetails } from 'screens/TransactionDetails';
import renderer from 'react-test-renderer';
import { Transaction, TxType } from 'models/transaction';
import { Currency, Wallet } from 'models/wallet';

it('Test tx details #1', () => {
    const tree = renderer
        .create(<TransactionDetails route={
            {
                params: {
                    transaction: new Transaction("Id#1", "Member#1", Currency.Polkadot, TxType.Sent, new Date("12-12-2020"), 111, 112),
                    wallet: new Wallet("Wallet#1", "Address#1", Currency.Polkadot, 100, 101)
                }
            }
        } />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('Test tx details #2', () => {
    const tree = renderer
        .create(<TransactionDetails route={
            {
                params: {
                    transaction: new Transaction("Id#2", "Member#2", Currency.Polkadot, TxType.Received, new Date("12-12-2020"), 211, 212),
                    wallet: new Wallet("Wallet#2", "Address#2", Currency.Kusama, 200, 201)
                }
            }
        } />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
