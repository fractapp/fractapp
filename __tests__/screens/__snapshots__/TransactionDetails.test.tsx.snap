// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test tx details #1 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 10,
        "borderWidth": 0.5,
        "justifyContent": "center",
        "marginTop": 20,
        "paddingBottom": 40,
        "paddingTop": 30,
        "width": "90%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/img/dot.png",
        }
      }
      style={
        Object {
          "borderRadius": 75,
          "height": 80,
          "overflow": "hidden",
          "width": 80,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Roboto-Regular",
          "fontSize": 16,
          "marginTop": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      Member#1
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Roboto-Regular",
          "fontSize": 40,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      -
      111
       
      DOT
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#888888",
              "fontSize": 25,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î¡¶
      </Text>
      <Text
        style={
          Object {
            "color": "#888888",
            "fontFamily": "Roboto-Regular",
            "fontSize": 18,
          }
        }
      >
        Success
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 30,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "color": "#888888",
            "fontFamily": "Roboto-Regular",
            "fontSize": 18,
            "marginLeft": 20,
            "marginRight": 20,
          },
          Object {
            "marginBottom": 10,
          },
        ]
      }
    >
      Write-off account
    </Text>
    <View
      style={
        Object {
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "height": 50,
              "marginBottom": 10,
              "marginTop": 10,
              "paddingLeft": "5%",
              "width": "95%",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/img/dot.png",
                }
              }
              style={
                Object {
                  "borderRadius": 75,
                  "height": 50,
                  "overflow": "hidden",
                  "width": 50,
                }
              }
            />
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column-reverse",
                  "marginLeft": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "height": 25,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "lineHeight": 18,
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    100
                     
                    DOT
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "lineHeight": 18,
                        },
                        Object {
                          "textAlign": "right",
                        },
                      ]
                    }
                  >
                    10100
                    $
                  </Text>
                </View>
              </View>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 15,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "lineHeight": 18,
                  }
                }
              >
                Wallet#1
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "color": "#888888",
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
              "marginLeft": 20,
              "marginRight": 20,
            },
            Object {
              "marginBottom": 5,
            },
          ]
        }
      >
        Date
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 18,
            "marginLeft": 20,
            "marginRight": 20,
          }
        }
      >
        2020-12-12
         
        0:00:00
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "color": "#888888",
                "fontFamily": "Roboto-Regular",
                "fontSize": 18,
                "marginLeft": 20,
                "marginRight": 20,
              },
              Object {
                "marginBottom": 5,
              },
            ]
          }
        >
          Fee
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "flex-start",
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        >
          112
           
          DOT
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test tx details #2 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#888888",
        "borderRadius": 10,
        "borderWidth": 0.5,
        "justifyContent": "center",
        "marginTop": 20,
        "paddingBottom": 40,
        "paddingTop": 30,
        "width": "90%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/img/dot.png",
        }
      }
      style={
        Object {
          "borderRadius": 75,
          "height": 80,
          "overflow": "hidden",
          "width": 80,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Roboto-Regular",
          "fontSize": 16,
          "marginTop": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      Member#2
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Roboto-Regular",
          "fontSize": 40,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      +
      211
       
      DOT
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#888888",
              "fontSize": 25,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î¡¶
      </Text>
      <Text
        style={
          Object {
            "color": "#888888",
            "fontFamily": "Roboto-Regular",
            "fontSize": 18,
          }
        }
      >
        Success
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 30,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "color": "#888888",
            "fontFamily": "Roboto-Regular",
            "fontSize": 18,
            "marginLeft": 20,
            "marginRight": 20,
          },
          Object {
            "marginBottom": 10,
          },
        ]
      }
    >
      Write-off account
    </Text>
    <View
      style={
        Object {
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Object {
              "height": 50,
              "marginBottom": 10,
              "marginTop": 10,
              "paddingLeft": "5%",
              "width": "95%",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/img/kusama.png",
                }
              }
              style={
                Object {
                  "borderRadius": 75,
                  "height": 50,
                  "overflow": "hidden",
                  "width": 50,
                }
              }
            />
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column-reverse",
                  "marginLeft": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "height": 25,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "lineHeight": 18,
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    200
                     
                    KSM
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontFamily": "Roboto-Bold",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "lineHeight": 18,
                        },
                        Object {
                          "textAlign": "right",
                        },
                      ]
                    }
                  >
                    40200
                    $
                  </Text>
                </View>
              </View>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 15,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "lineHeight": 18,
                  }
                }
              >
                Wallet#2
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "color": "#888888",
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
              "marginLeft": 20,
              "marginRight": 20,
            },
            Object {
              "marginBottom": 5,
            },
          ]
        }
      >
        Date
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 18,
            "marginLeft": 20,
            "marginRight": 20,
          }
        }
      >
        2020-12-12
         
        0:00:00
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "color": "#888888",
                "fontFamily": "Roboto-Regular",
                "fontSize": 18,
                "marginLeft": 20,
                "marginRight": 20,
              },
              Object {
                "marginBottom": 5,
              },
            ]
          }
        >
          Fee
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "flex-start",
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        >
          212
           
          DOT
        </Text>
      </View>
    </View>
  </View>
</View>
`;
